Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceManagmentAndroidDeviceOwnerEnrollmentProfile/MSFT_IntuneDeviceManagmentAndroidDeviceOwnerEnrollmentProfile.psm1

```diff
@@ -0,0 +1,699 @@
+        [System.String]
+        $CertificateThumbprint,
+
+        [Parameter()]
+        [Switch]
+        $ManagedIdentity,
+
+        [Parameter()]
+        [System.String[]]
+        $AccessTokens
+    )
+
+    $ConnectionMode = New-M365DSCConnection -Workload 'MicrosoftGraph' `
+        -InboundParameters $PSBoundParameters
+
+    #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
+
+    #region Telemetry
+    $ResourceName = $MyInvocation.MyCommand.ModuleName.Replace('MSFT_', '')
+    $CommandName = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    try
+    {
+        $Script:ExportMode = $true
+        ##TODO - Replace Get-Cmdlet by the cmdlet to retrieve all instances
+        [array] $Script:exportedInstances = Get-MgAndroidDeviceOwnerEnrollmentProfile
+         -ErrorAction Stop
+
+        $i = 1
+        $dscContent = ''
+        if ($Script:exportedInstances.Length -eq 0)
+        {
+            Write-Host $Global:M365DSCEmojiGreenCheckMark
+        }
+        else
+        {
+            Write-Host "`r`n" -NoNewline
+        }
+        foreach ($config in $Script:exportedInstances)
+        {
+            $displayedKey = $config.Id
+            Write-Host "    |---[$i/$($Script:exportedInstances.Count)] $displayedKey" -NoNewline
+            $params = @{
+            Id                    = $androidDeviceOwnerEnrollmentProfile.Id
+            DisplayName           = $androidDeviceOwnerEnrollmentProfile.DisplayName
+            AccountId             = $androidDeviceOwnerEnrollmentProfile.AccountId
+            ConfigureWifi         = $androidDeviceOwnerEnrollmentProfile.ConfigureWifi
+            CreatedDateTime       = $androidDeviceOwnerEnrollmentProfile.CreatedDateTime
+            Description           = $androidDeviceOwnerEnrollmentProfile.Description
+            EnrolledDeviceCount   = $androidDeviceOwnerEnrollmentProfile.EnrolledDeviceCount
+            EnrollmentMode        = $androidDeviceOwnerEnrollmentProfile.EnrollmentMode
+            EnrollmentTokenType   = $androidDeviceOwnerEnrollmentProfile.EnrollmentTokenType
+            EnrollmentTokenUsageCount = $androidDeviceOwnerEnrollmentProfile.EnrollmentTokenUsageCount
+            IsTeamsDeviceProfile  = $androidDeviceOwnerEnrollmentProfile.IsTeamsDeviceProfile
+            LastModifiedDateTime  = $androidDeviceOwnerEnrollmentProfile.LastModifiedDateTime
+            QrCodeContent         = $androidDeviceOwnerEnrollmentProfile.QrCodeContent
+            QrCodeImage           = $androidDeviceOwnerEnrollmentProfile.QrCodeImage
+            RoleScopeTagIds       = $androidDeviceOwnerEnrollmentProfile.RoleScopeTagIds
+            TokenCreationDateTime = $androidDeviceOwnerEnrollmentProfile.TokenCreationDateTime
```

Comment:
Again, look at removing properties that can't be set. In the case of the export, you can only specify Name and ID here.